test/DoctrineBatchUtilsTest/BatchProcessing/SelectBatchIteratorAggregateTest.php

```diff
@@ -115,7 +115,7 @@ public function testIterationWithSuccessfulReFetches(): void
             ['Yadda', ['id' => 123], $freshObjects['foo']],
             ['Yadda', ['id' => 456], $freshObjects['bar']],
         ]);
-        $this->entityManager->expects(self::at(2))->method('clear');
+        $this->entityManager->expects(self::once())->method('clear');
```

Comment:
I've mentioned this previously: this `clear()` must be called after the other invocations

Reply:
I have fixed this test but i think it is really messy, have you a more clean solution ?

Reply:
It's messy, but it is indeed required. The best we can do is isolating the stub creation to a `private` method.

Further refactoring can be deferred to a later point in time: for now, this (IMO) works well enough.